apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/db/ShardDataSourceCreateHelperTest.java

```diff
@@ -0,0 +1,62 @@
+package com.apollocurrency.aplwallet.apl.core.db;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+
+import com.apollocurrency.aplwallet.apl.core.app.EpochTime;
+import com.apollocurrency.aplwallet.apl.extension.DbExtension;
+import com.apollocurrency.aplwallet.apl.util.NtpTime;
+import com.apollocurrency.aplwallet.apl.util.injectable.PropertiesHolder;
+import org.jboss.weld.junit.MockBean;
+import org.jboss.weld.junit5.EnableWeld;
+import org.jboss.weld.junit5.WeldInitiator;
+import org.jboss.weld.junit5.WeldSetup;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@EnableWeld
+class ShardDataSourceCreateHelperTest {
+
+    @RegisterExtension
+    static DbExtension extension = new DbExtension();
+
+    @WeldSetup
+    public WeldInitiator weld = WeldInitiator.from(
+            PropertiesHolder.class, ShardRecoveryDaoJdbcImpl.class, EpochTime.class)
+            .addBeans(MockBean.of(extension.getDatabaseManger(), DatabaseManager.class))
+            .addBeans(MockBean.of(mock(NtpTime.class), NtpTime.class))
+            .build();
+
+    private ShardDataSourceCreateHelper createHelper;
+
+    @AfterAll
+    static void tearDown() {
+        extension.getDatabaseManger().shutdown();
```

Comment:
extension will shutdown automatically. Delete this pls

Reply:
Done